influxql/query_executor.go

```diff
@@ -392,13 +402,33 @@ LOOP:
 	}
 }
 
+// Determines if the QueryExecutor will recover any panics or let them crash
+// the server.
+var willCrash bool
+
+func init() {
+	var err error
+	if willCrash, err = strconv.ParseBool(os.Getenv(PanicCrashEnv)); err != nil {
+		willCrash = false
+	}
+}
+
 func (e *QueryExecutor) recover(query *Query, results chan *Result) {
 	if err := recover(); err != nil {
+		atomic.AddInt64(&e.stats.RecoveredPanics, 1) // Capture the panic in _internal stats.
 		e.Logger.Error(fmt.Sprintf("%s [panic:%s] %s", query.String(), err, debug.Stack()))
 		results <- &Result{
 			StatementID: -1,
 			Err:         fmt.Errorf("%s [panic:%s]", query.String(), err),
 		}
+
+		if willCrash {
+			e.Logger.Error(fmt.Sprintf("\n\n=====\nAll goroutines now follow:"))
+			buf := make([]byte, 1<<16)
+			stackSize := runtime.Stack(buf, true)
+			e.Logger.Error(fmt.Sprintf("%s\n", string(buf[0:stackSize])))
+			panic(err) // Ensure any defers run
```

Comment:
@jsternberg I thought about this but I wasn't aware of a way to do that. I think once `recover()` has been executed you're done with the original panic and you can't "rethrow" it can you? Given that I was left with `panic(err)` or `panic(nil)` for example.